sql/core/src/test/scala/org/apache/spark/sql/SQLQueryTestSuite.scala

```diff
@@ -308,6 +309,7 @@ class SQLQueryTestSuite extends QueryTest with SharedSparkSession {
         localSparkSession.conf.set(SQLConf.CROSS_JOINS_ENABLED.key, true)
         localSparkSession.conf.set(SQLConf.ANSI_SQL_PARSER.key, true)
         localSparkSession.conf.set(SQLConf.PREFER_INTEGRAL_DIVISION.key, true)
+        localSparkSession.conf.set(SQLConf.FAIL_ON_INTEGRAL_TYPE_OVERFLOW.key, true)
```

Comment:
instead of setting it always to true, I think we should set it to both true and false in the tests which are relevant.

Reply:
I prefer to enable it for all pgSQL tests:
1. If we are going to have one flag for ANSI mode such as https://github.com/apache/spark/pull/25693, then we should enable all the ANSI features here.
2. If we have to isolate this flag, then we need to isolate `SQLConf.ANSI_SQL_PARSER` as well.

Reply:
in this way we are running all our tests only in one of the two modes (the non-default one moreover), so we are not ensuring the behavior on both modes. I don't think this is a good idea. I think that when we will have the ANSI feature, we will run these tests both with ANSI enabled and not. Otherwise we would not provide proper coverage.

Reply:
Make sense. I think the cause of the issue is test coverage.
Testing both modes for pgSQL tests seems great. The total time of pgSQL tests is around 7 minutes in my local setup, and `org.apache.spark.sql.SQLQueryTestSuite` is executed in parallel(see SparkBuild.scala).